controller/lib/serdes/controller_data_packet_encoder.cpp

```diff
@@ -0,0 +1,30 @@
+#include "serdes.h"
+#include <pb_common.h>
+#include <pb_encode.h>
+bool serdes_encode_status_packet(uint64_t time, float pressure, float volume,
+                                 float flow, uint8_t *tx_buffer,
+                                 uint8_t tx_buffer_len, size_t *encoded_len) {
```

Comment:
This is an obvious change I consciously withheld to avoid leaking network_protocol dependency all over the system. For now. Now the only place they leaked is command/alert packet handlers (to be migrated asap). 

This is related to another global discussion I would like to have - the model. I have already created model.h and moved some type definitions, that are used in parameters.h to it. If we look at parameters.h we can see a lot of space for improvement - at least grouping parameters to related structs, like PID parameters, parameters related to outer control loop, general low level stuff, etc. Once we do this work, model-dictated ControllerStatus type will pop up.

I would like not to leak nanopb types to the system - they are specifically designed to work with nanopb machinery, are auto-generated, etc. I would use types coming from model here and do translation to nanopb.

Reply:
> I would like not to leak nanopb types to the system - they are specifically designed to work with nanopb machinery, are auto-generated, etc. I would use types coming from model here and do translation to nanopb.

I'm trying to see what benefit having this layer of indirection would get us.  Under what circumstances do you imagine the model struct and nanopb structs being different?

Reply:
I just have this rule of thumb - network stuff is different than a model, just picked it up over the years. It pays off having adapters on system boundaries. Like if we decide to do away with nanopb. 
But I agree, having arguments listed like that is ugly too, also error prone (although it will be likely used in a single place)

It's a dilemma - either litter the system with network-protocol dependency or drag ControllerStatus into the model which would be perfect, but then we will have to manually edit nanopb generated files and delete those types which is ugly too.

Reply:
> It's a dilemma

Yeah, Google has this dilemma too, because we use protos everywhere...  Protos aren't quite the same as network structs, but I hear the worry about being able to change things.  It can become annoying to deal with proto objects when you really want a "nice C++ object".

My feeling is, software is changeable.  Like, if we decide to drop nanopb, we can declare structs that look the same as the generated code, if we want.  Or if we decide that the proto struct that we're passing around isn't working for us, we can introduce a layer of indirection like you described.  But there's a cost to every layer of indirection we add (in this case, keeping it in sync with the proto), and so why pay that cost until we're getting a benefit from it?

Basically, https://en.wikipedia.org/wiki/YAGNI.

WDYT?

Reply:
separate variables replaced with ControllerStatus